asm_utils/src/data_parser.rs

```diff
@@ -14,76 +18,131 @@ pub enum DataValue {
 
 impl DataValue {
     /// Returns the size of the value in bytes.
-    pub fn size(&self) -> usize {
+    ///
+    /// The address is necessary because the size of the alignment padding
+    /// depending on what address it is defined on.
+    pub fn size(&self, from_addr: usize) -> usize {
         match self {
             DataValue::Direct(data) => data.len(),
             DataValue::Zero(length) => *length,
+            DataValue::Alignment(bytes) => alignment_size(from_addr, *bytes),
```

Comment:
so the alignment directive essentially creates a padding, right?

Reply:
Not necessarily. Only if needed.